pkg/controllers/machine/disruption/drift.go

```diff
@@ -78,3 +78,31 @@ func (d *Drift) Reconcile(ctx context.Context, _ *v1alpha5.Provisioner, machine
 	// Requeue after 5 minutes for the cache TTL
 	return reconcile.Result{RequeueAfter: 5 * time.Minute}, nil
 }
+
+// isDrifted will increase in scope to include more fields of the provisioner
+func (d *Drift) isDrifted(ctx context.Context, provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) (bool, error) {
+	cloudProviderDrifted, err := d.cloudProvider.IsMachineDrifted(ctx, machine)
+	if err != nil {
+		return false, err
+	}
+	staticFieldsDrifted := areStaticFieldsDrifted(provisioner, machine)
+
+	return cloudProviderDrifted || staticFieldsDrifted, nil
+}
+
+// Candidates for Static Fields
+// //////////////////////
+// Taints
+// Labels
+// Annotations
+// Kubelet Configuration
+// Startup Taints
+// //////////////////////
```

Comment:
nit: Probably better to just link the Karpenter docs here so that it stays up to date.